data-access-layer/service-agent/Agent.js

```diff
@@ -196,27 +197,45 @@ class Agent extends HttpClient {
     // In case agent passwords are being updated as part of this update
     return this
       .getHost(ips, 'lifecycle.preupdate')
-      .then(ip => this.post(ip, 'lifecycle/preupdate', context, 200, agentCredsBeforeUpdate));
+      .then(ip => this.post(ip, 'lifecycle/preupdate', context, CONST.HTTP_STATUS_CODE.OK, agentCredsBeforeUpdate));
   }
 
   createCredentials(ips, parameters, preBindResponse) {
     const body = {
       parameters: parameters,
       actions: preBindResponse
     };
-    return this
-      .getHost(ips, 'credentials')
-      .then(ip => this.post(ip, 'credentials/create', body, 200));
+    return Promise
+      .any(_.map(ips, ip => utils.retry(() => this
```

Comment:
Promise.any wrapping retry not a good combination. Since retry happens in background even though one of the promises are resolved retry on unsupported agent also will continue to happen.

Why not wrap the existing code in util.retry like the below?
```
 return utils.retry(() => { 
      return this
      .getHost(ips, 'credentials')
      .then(ip => this.post(ip, 'credentials/create', body, 200))},
            maxAttempts: 2,
            timeout: CONST.AGENT.OPERATION_TIMEOUT_IN_MILLIS,
            predicate: (err) => (err instanceof errors.Timeout)
          }); 
;
```
The above code does a retry of the entire stuff again in case of timeout error. In all other cases you should not retry. The below code will end up retrying even when there is legitigmate errors being thrown from agent.

Reply:
removed retry from here.